tools/buildmgr/cbuildgen/src/CMakeListsGenerator.cpp

```diff
@@ -305,17 +355,15 @@ bool CMakeListsGenerator::GenBuildCMakeLists(void) {
   // Set global flags
   cmakelists << "# Global Flags" << EOL << EOL;
 
-  bool asflags = as_file_specific_flags || as_group_specific_flags;
-  bool ccflags = cc_file_specific_flags || cc_group_specific_flags;
-  bool cxxflags = cxx_file_specific_flags || cxx_group_specific_flags;
-
+  bool target_options = !m_optimize.empty() || !m_debug.empty() || !m_warnings.empty();
   for (auto list : asFilesLists) {
     if (!list.second.empty()) {
       string prefix = list.first;
       cmakelists << "set(CMAKE_" << prefix << "_FLAGS \"${" << prefix << "_CPU}";
       if (!m_byteOrder.empty()) cmakelists << " ${" << prefix << "_BYTE_ORDER}";
-      if (!m_definesList.empty()) cmakelists << " ${" << prefix << "_DEFINES} ${";
-      cmakelists << prefix << "_FLAGS}";
+      if (!m_definesList.empty()) cmakelists << " ${" << prefix << "_DEFINES}";
+      if (!specific_options && target_options) cmakelists << " ${ASM_OPTIONS_FLAGS}";
```

Comment:
do you think this should be AS_OPTIONS_FLAGS ?

Reply:
I think it should have the assembler dialect `prefix`:

```suggestion
      if (!specific_options && target_options) cmakelists << " ${" << prefix << "_OPTIONS_FLAGS}";
```
For IAR it corresponds to `ASM_OPTIONS_FLAGS`, but other compilers may have different dialects.